### PR TITLE
Added functionality for flaky tests, more granular step definitions and upload of evidence 

### PR DESCRIPTION
**Pull request by Niklas-Back**
This PR introduces the ability to choose what evidence is to be uploaded due to Xrays 100MiB limit. It also adds the possibility to detect flaky tests and it removes failed reruns from the json import file (also because of the 100MiB limit) When debug option is enabled there will be a new file created, xray-payload-debug.json that contains the report file before duplicates has been removed and flaky tests has been detected. I also added the option to set summary

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -19,12 +19,20 @@ class XrayReporter implements Reporter {
...
+  private uploadTrace: boolean | undefined;
+  private uploadVideo: boolean | undefined;
+  private stepCategories = ['expect', 'pw:api', 'test.step'];
 
   constructor(options: XrayOptions) {
     this.options = options;
     this.help = new Help(this.options.jira.type);
     this.xrayService = new XrayService(this.options);
     this.totalDuration = 0;
+    this.upploadScreenShot = options.uploadScreenShot;
```
Comment (inluxc) [inline src/index.ts:32]: Fix typo: upploadScreenShot => uploadScreenShot

File: src/index.ts [hunk 2]
```diff
@@ -82,7 +91,8 @@ class XrayReporter implements Reporter {
       } else {
         await Promise.all(
           result.steps.map(async (step) => {
-            if (step.category != 'hook') {
+            //if ( step.category == 'test.step') {
```
Comment (inluxc) [inline src/index.ts:94]: Please remove extra commented code

**Review by inluxc — CHANGES_REQUESTED** [review]
Can you check the comments.
Thanks for the PR

**Comment by NiklasBack** [comment]
I'll just run the modified code in our CI

**Review by inluxc — APPROVED** [review]

**Pull request merged**